tests/semanticdb/metac.expect

```diff
@@ -2324,6 +2374,7 @@ Occurrences:
 [16:29..16:32): ??? -> scala/Predef.`???`().
 [17:6..17:8): m7 <- example/Methods#m7().
 [17:9..17:10): U <- example/Methods#m7().[U]
+[17:10..17:10): <- example/Methods#m7().(evidence$1)
```

Comment:
we still don't want evidence parameters (desugared from context bounds) showing as a definition in the occurrences section I believe, as they are not visible in the source code, what do you think @tgodzik?

Reply:
Yeah, that should ideally be on the synthetics section

Reply:
Ah right, I will exclude them from `occurrence` while keeping them in SymbolInformation section :+1:

Reply:
done https://github.com/lampepfl/dotty/pull/12885/commits/8801a69d1f382f688fdfca1ba02ca12874191db9